torchtune/modules/loss/cross_entropy_loss.py

```diff
@@ -143,3 +143,95 @@ def forward(
             return total_loss
         else:
             return total_loss / total_elements
+
+class LigerFusedCrossEntropyLoss(nn.Module, SFTLoss):
+    """Memory efficient Cross-entropy loss that uses fused CUDA kernels to compute the loss.
+    Combines the linear projection with the cross-entropy calculation for better performance
+    and memory efficiency.
+
+    Linear cross entropy is computed in a single fused operation. You need to skip the final 
+    projection layer in your model and pass it to the loss instead. You can setup the loss 
+    with the model as shown below.
+
+    >>> model = Transformer(...)
+    >>> loss = LigerFusedCrossEntropyLoss(...)
+    >>> loss.set_model_output(model)
+    """
+
+    def __init__(
+        self,
+        ignore_index: int = -100
+    ):
+        """Initialize the LigerFusedCrossEntropyLoss.
+
+        Args:
+            ignore_index (int): Index to ignore in the target tensor. Default is -100.
+        """
+        super().__init__()
+        self.linear_projection = None
+        self.ignore_index = ignore_index
+
+    def set_model_output(self, model: nn.Module) -> None:
+        """Modify model output to match the expected input for the loss function.
+        
+        Args:
+            model (nn.Module): The model whose output layer will be used for the loss computation.
+        """
+        model.skip_output_layer = True
+        self.linear_projection = model.output
```

Comment:
Since this loss requires that the linear_projection contains a weight and bias parameter. It might be a good idea to check for that here and raise an Exception if they're missing or if there are other parameters.